lib/main.js

```diff
@@ -30,19 +30,29 @@ bedrock.events.on('bedrock.configure', function applyConfig() {
     throw new Error('"bedrock-config-yaml" must be the last import.');
   }
 
-  _applyConfig({type: 'app'});
+  _applyConfig({configType: 'app'});
 });
 
-function _applyConfig({type}) {
+function _applyConfig({configType}) {
+  logger.debug(`Attempting to apply the "${configType}" configuration.`);
   // attempt to load the YAML file specified by the config
-  const configFile = path.join(cfg[type].path, cfg[type].filename);
-  if(fs.existsSync(configFile)) {
-    logger.debug(`"${type}" configuration found "${configFile}".`);
-    const configYaml = jsYaml.safeLoad(fs.readFileSync(configFile, 'utf8'));
-
-    // params: deep, target, source
-    extend(true, config, configYaml);
-  } else {
-    logger.debug(`"${type}" configuration not found "${configFile}".`);
+  const types = ['combined', configType];
+  for(const type of types) {
+    const configFile = path.join(cfg[type].path, cfg[type].filename);
+    if(fs.existsSync(configFile)) {
```

Comment:
Ideally, `_applyConfig` would be an async function and this would use `await` on the promise version of `fs.exists`. Seeing as the original version was still using blocking IO, I wouldn't hold up the PR for this. I'll create an issue out of this comment.